### ISSUE TITLE
Add custom licences and store in database

### ISSUE THREAD

**Issue by EricGuic** [Chocobozzz/PeerTube#6040]
### Describe the problem to be solved

For some administrators (like me, working for a public organization), the Creative Commons native licences are not enough.
I would like to be able to add some custom licences, and to attach some url to it (redirecting the user to the legal explanations).

This could be achieve with a plugin, but for the new licences to be broadcasted to other federated instances, their admin should also install the same plugin, with the same parameters (the custom licences).

So maybe a better solution could be achieve attaching the licence name and url to the video in database.

### Describe the solution you would like

After a first discussion (in french) here : https://framacolibri.org/t/ajout-de-licences-personnalisables/19584
this is what it could looks like : 

- Store the license of the video in peertube's database

- Store also an url linked to each custom licence (for example to redirect the user on click to the specific page displaying the legal explanations)

- Add section in the webUI, allowing the administrator to see all the licences he wants to offer to his users : the native Creative Commons licences, and if needed, the custom one he've created.

So to add a custom licence, the admin should fill two fileds, the first with the displayed name of the licence, and a second one with the url of the legal explanations. Once complete and saved, this new custom licence will be available in the dropdown menu of the video parameters.

JohnLivingston has made a usefull proposal : to write also a Fediverse Enhancement Proposal so this feature could be supported at a higher level by ActivityPub.

**Comment by Chocobozzz** [comment]
Seems an interesting idea. We just won't have time to implement this feature in 2025 I think.

**Comment by EricGuic** [comment]
Hi Chocobozzz, I try again for the 2026 roadmap : do you still think there could be some improvments with this idea, and do you plan to work on it ?
Best regards,
Eric

**Comment by Chocobozzz** [comment]
No sorry it's not on our roadmap. Can you detail which licence you would like to introduce?

**Comment by EricGuic** [comment]
In my case (public organization in education), the majority of our ressources are produced via partnership, and some of them use protected materials (eg : musics or paintings reproduced in a video about Art History).
So we can't use Creative Common licences for example.

So in my case again (but I think most organizations switching to Peertube face the same problem), we have our own licence/CGU/CGV. Being able to point an url displaying it (elswhere or on a custom page inside peertube) would solve our problem.
But only on our instance.
As explained previously, to solve it when our videos are played on another federated instance, the licence should "follow" the video.

Do I make myself clear ? Don't hesitate to reach me directly to discuss it (in french :-)

I'm trying for a long time now to push my organization to participate in this development (by freeing dev time, or funding it), but Peertube isn't a priority here, and I feel lonely supporting it. I manage to drive the developpment of PMU (Peertube Mass Uploader) and to share the code, but nothing more since. The best I can do right now is providing detailed feedbacks and reaching you for those which seems to have the potential to address other's needs and facilitate Peertube adoption.

**Comment by EricGuic** [comment]
Hi, just a little message to see if maybe other users have reported similar needs for a better licence feature ?
Maybe did you take some time to think about it, and could provide your thoughts about the possibility to add it to the roadmap ?
Best regards, Eric

**Comment by EricGuic** [comment]
Hi, just wondering about the next roadmap : is there a possibility to implement this improved licence feature ?
(In my organization, I've been explained that this is maybe the first reason that prevent a larger adoption, for leagal reason of course. So I guess maybe other non-individual users could benefit from it).
Best regards, Eric.

**Comment by Chocobozzz** [comment]
No sorry, this feature is still not in our roadmap because we think the plugin API can be used to inject the licenses you want.
Can you explain why you don't want to develop a plugin? Is it because other instances won't be able to recognize your custom licence if they don't install it?

**Comment by EricGuic** [comment]
Yes, that's exactly the main reason (I will continue in French to make myself clear) : 

L'idée est de développer un réseau d'instances Peertube, hébergées par différents acteurs du monde éducatif (nous opérateurs de l'état, le ministère avec ses instance _Apps.education_, les académies, les Environnements Numériques de Travail des établissements scolaires portés par les régions ou départements...). Or c'est impossible de coordonner tous ces acteurs disparates pour installer à chaque fois le plugin, le configurer et le mettre à jour dans le temps.

Je constate déjà aujourd'hui la limite du principe de plugin, avec celui intitulé "_categorie_" développé par la DNE (Direction du Numérique Educatif) : via ce plugin, l'administrateur d'une instance attribue un numéro à de nouvelles catégories qu'il peut ajouter. La DNE nous a fourni une liste, que nous avons appliquée sur notre instance. Mais ces nouvelles catégories personnalisées ne sont pas reconnues par les autres instances qui n'ont pas installé le plugin, ou qui sont sur une ancienne version de la numérotation.
Donc la logique de plugin attend vite ses limites lorsqu'on veut aller plus loin qu'une instance isolée, et vraiment jouer la carte de la fédération.

Donc à mon sens, les améliorations de ces points (les licences comme les catégories) devraient être intégrées au code source pour améliorer l'adoption de Peertube chez les acteurs institutionnels.

Mais je devine bien le travail conséquent, et je connais les forces de Framasoft sur ce projet.
